compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -6873,16 +6873,22 @@ private STNode createResourcePathNodeList(List<STNode> pathElementList) {
      *
      * @return Parsed node
      */
-    private STNode parseResourcePathSegment() {
+    private STNode parseResourcePathSegment(boolean isFirst) {
```

Comment:
```suggestion
    private STNode parseResourcePathSegment(boolean isFirstSegment) {
```
I think it is more meaningful.